### ISSUE TITLE
Limit specific system formatting options to selected system

### ISSUE THREAD

**Issue by misthero** [naturalcrit/homebrewery#3573]
### Your idea:

Since hombrewery already let you check what system are we homebrewing for, would be nice to avoid having 5e graphics, css, fonts and formatting when selecting other system, I know the checkbox is just for indexing right now and 5e is the most used and no other system is really supported... but would be nice not needing to override everything when creating hombrew for something different from 5e.

![image](https://github.com/user-attachments/assets/50f752e4-654a-4f0b-a11f-3d65d2241480)

Just my 2 cents.

Thx for the hard work.

**Comment by calculuschild** [comment]
Something like this has actually been on the site, hidden, for about a year. Just too many other requests and higher priority bugs to actually finish it off.

BUT, we have a PR right now that makes the final touches and will let you make custom themes and share them much more easily. Crossing fingers we can release it very soon....

**Comment by dbolack-ab** [comment]
It's also important to note that the System checkboxes were not used for theme keying and were instead a predecessor to the searchable tags.

**Comment by misthero** [comment]
> It's also important to note that the System checkboxes were not used for theme keying and were instead a predecessor to the searchable tags.

yep, I'm a long time user of homebrewery and as I said:

> I know the checkbox is just for indexing right now and 5e is the most used and no other system is really supported

In my opinion, it makes sense to have a "default style" applied based on the selected system, but that's just a suggestion. 

Over the past few years, I've created many 5e brews, some of which are on DMsGuild with thousands of downloads, and I've never felt the need for such a feature. However, when I started making PF2 brews and checked the Pathfinder checkbox, I was disappointed to see the 5e styling everywhere. 

Even a basic "blank/general style" would be helpful to reduce the initial work when starting with a different system.

**Comment by misthero** [comment]
> With custom themes enabled, and the static blank theme too, i'd say this just comes down to: do we delete the system UI and leave it to tags? or not?
Here's a revised version of your comment:

I was thinking about this a few days ago, and I believe it depends on your future plans. Custom themes is great, but if the goal is to expand on it, the system UI could evolve into something that categorizes themes and brews.

In a hypothetical future where you have dozens of themes for different systems, this could help narrow down the themes to choose from. For example, if I only want to see themes for 5e, I could select 5e as my preferred system and browse the available themes. This might be even more useful when theme sharing is live, as not every user may give their theme a meaningful name.

Additionally, if a brew search feature is introduced, the system tag could be helpful for that as well.

However, if none of the above is planned, then I think the system tag could be removed.

What do you think?

**Comment by 5e-Cleric** [comment]
Appending themes to systems is an interesting idea, but i don't think we need to be that restrictive.

**Comment by ericscheid** [comment]
Themes and System are different things. Though they often overlap, they don't map usefully.

`AD&D 2e` is a system,  `Planescape` is a theme, `Dark Sun` is a theme, `AD&D 2e` (vanilla) is a theme.

The tags UI was built to accomodate deprecating Systems UI, and is expandable as needed by users. The Systems UI is a static list of checkboxes.

> do we delete the system UI and leave it to tags?

That was always the plan. The only thing holding that back is the lack of a deployed datalist UI to show tags are the way forward.

**Comment by dbolack-ab** [comment]
> Additionally, if a brew search feature is introduced, the system tag could be helpful for that as well.

This should be largely resolved by the Vault features, presuming users label their brews usefully.

**Comment by 5e-Cleric** [comment]
> > Additionally, if a brew search feature is introduced, the system tag could be helpful for that as well.
> 
> This should be largely resolved by the Vault features, presuming users label their brews usefully.

Yes, once Vault is up i will work on that feature, and should be somwehat simple

**Comment by misthero** [comment]
> Themes and System are different things. Though they often overlap, they don't map usefully.
> 
> `AD&D 2e` is a system, `Planescape` is a theme, `Dark Sun` is a theme, `AD&D 2e` (vanilla) is a theme.

I agree with your point that themes and systems are distinct. However, this doesn't conflict with the idea that multiple themes can fit within a single system, and that the system tag could be a useful filter for finding the appropriate theme.

For example, if I’m looking for a theme for AD&D 2e, I could filter by the system tag “AD&D 2e” and get the following results:

1.    AD&D 2e (vanilla)
2.    Planescape
3.    Dark Sun

This gives me three choices, allowing me to select the best theme for my AD&D 2e homebrew.

Of course, this approach would be more beneficial if we have a large number of themes available. I'm not trying to push anything, just sharing ideas for discussion.

**Comment by ericscheid** [comment]
A brew can have multiple tags, including being tagged by more than variation of a base system.

There is currently no `AD&D` system option. Each "system" option needs to be first added by the developers.

This is why we implemented tags, and especially the `system:` tag-prefix, to allow for multiple unforseen values.

**Comment by dbolack-ab** [comment]
> Yes, once Vault is up i will work on that feature, and should be somwehat simple

Is vault in a place where we can consider removing the system boxes/migrating them to tags/etc. ?

**Comment by 5e-Cleric** [comment]
Vault should have nothing to do with tags and systems, since none of them is searchable as of yet.

I'm gonna close this up and reopen an issue about removing systems entirely in favor of tags. And closing this, as merging systems into themes is simply a bad idea.
